src/Schema/OracleSchemaManager.php

```diff
@@ -320,6 +320,109 @@ private function getQuotedIdentifierName($identifier)
         return $identifier;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        if ($currentDatabase === null) {
+            throw new Exception('Invalid database name.');
```

Comment:
Unless there's a way to test this exception, it should be an assertion. While it's possible for other platforms (e.g. MySQL), I cannot of a way to make the current user unavailable on an Oracle connection.

Reply:
Done